### DISCUSSION TITLE
Latent mode with LoRA applied to all regions?

### DISCUSSION THREAD

**Discussion by catboxanon** [hako-mikan/sd-webui-regional-prompter#186] [Q&A]
As far as I know, latent mode currently applies LoRA by region in the order they're specified in the prompt. [This is the use case that's mentioned in the README](https://github.com/hako-mikan/sd-webui-regional-prompter#latent). Is there instead a way to apply all LoRA in the prompt to all regions specified via this extension? This is supported for attention mode (as LoRA separation can't be done at all in this mode), but I don't believe it is for latent mode.

The [two-shot](https://github.com/opparco/stable-diffusion-webui-two-shot) extension supports this since it isn't concerned about LoRA separation, but it also doesn't have the many other features of this extension (regions by inpaint, regions by prompt, etc). [This discussion](https://github.com/hako-mikan/sd-webui-regional-prompter/discussions/150) seems to imply attention mode is preferred but I don't believe this to be true, as the two-shot extension has worked perfectly fine with multiple LoRAs, and I'd go as far to say it works better in some scenarios since the latent space is larger than the attention space, [which is even mentioned in this extension's README](https://github.com/hako-mikan/sd-webui-regional-prompter#accuracy).

**Comment by Symbiomatrix** [comment]
Main issue with latent is time consumption - it grows linearly with the number of regions, whereas attention doesn't (or didn't, I'm out of touch on the latest webui shenanigans), hence my advice yonder. In terms of output quality, I concur that it's situational.
If you want, you can apply a lora to the base clause which is weighted in, or to the common clause which should be identical to a global lora; what latent mode does is switch between lora strengths whilst going through the network buildup, and then applies each conditioning matrix to its corresponding region via a filter. If the strengths are the same, the condition's matrix is modified equally. The one difference is, as I recall, you might need to set the negative textencoder / unet values to 1 to make it truly identical.
You say the two shot extension supports it - does it actually still work on the latest webui? That particular extension has been abandoned for 6 months and there have been several breaking changes for us, including a major overhaul to the lora system in march, so that's pretty astounding if it does.

**Reply by catboxanon** [reply]
> does it actually still work on the latest webui

It does, just tried it out a few days ago on dev. It doesn't touch LoRA code at all so that's likely why it works fine with LoRAs. It hijacks the composable diffusion code which hasn't been touched in months. Regional Prompter also produces identical results to it (without LoRAs of course). 

I'm pretty sure I already tried applying a single LoRA to the base clause and it still only affected a single region. I can try again in a bit tho if you do think it's supposed to work.
